design/fast_finality.md

```diff
@@ -0,0 +1,103 @@
+# Fast finality design
+
+## Scope
+
+The introduction of a single sequencer to improve cost-per-transaction during the bootstrapping phase, as 
+described in the [Bootstrapping Strategy design doc](./Bootstrapping_strategy.md).
+
+## Requirements
+
+* Finality
+  * Transaction *soft* finality (finality guaranteed by the sequencer) is achieved in under one second
+  * There is eventual transaction *hard* finality (finality guaranteed by the L1)
+  * The sequencer achieves hard finality on the agreed cadence
+  * The sequencer is not able to "rewrite history" (or is strongly disincentivised from doing so), even for soft-final
+    transactions (e.g. to perform front-running)
+* Censorship resistance
+  * There is an "inbox" allowing nodes to force the inclusion of transactions on the L1
+  * The sequencer must distribute all light batches to all nodes
+* Cost
+  * L1 transaction costs can be driven arbitrarily low, at the expense of extending the hard-finality window
+* User/dev experience
+  * The responses to RPC calls reflect the soft-finalised transactions, and not just the hard-finalised transactions
+* Operations
+  * The sequencer is highly available
+
+## Assumptions
+
+* There is a single aggregator that is also the genesis node. This node is known as the *sequencer*
+* The enclave has a start-up delay that exceeds the production time for light batches (see below)
+
+## Constraints
+
+* It must be possible to only publish rollups to the L1 every `x` blocks, where `x` >> 1
+* It must be possible to produce light batches (see below) at a higher frequency than L1 blocks
+
+## Design
+
+### Node start-up
+
+At start-up, the host checks if one of the following applies:
+
+* They are the genesis node and are an aggregator
+* They are *not* the genesis node and are *not* an aggregator
+
+If neither of these conditions is met, the host shuts down.
+
+The identity of the sequencer is listed in the management contract on the L1. This allows other nodes to verify that 
+the rollups are created by the sequencer.
+
+### Production of light batches
+
+On each block, the sequencer's host feeds a set of transactions to the enclave. The enclave responds by creating a 
```

Comment:
this has to change. There will be multiple LBs generated between 2 successive blocks.

Q: Do the LBs still need to be linked to the latest block that was fed into the enclave?
Pros/Cons

Reply:
Oops, left this bit about "each block" in by accident.